src/openforms/formio/migration_converters.py

```diff
@@ -45,10 +45,50 @@ def move_time_validators(component: Component) -> bool:
     return True
 
 
-# Keys are component["type"] values, values are dicts keyed by unique converter
-# identifier and the function to do the actual conversion.
+def alter_prefill_default_values(component: Component) -> bool:
+    """A converter that replaces ``prefill`` dict values from ``None`` to an empty string."""
+    if "prefill" not in component:
+        return False
+
+    altered = False
+
+    prefill_plugin = component["prefill"]["plugin"]
+    prefill_attribute = component["prefill"]["attribute"]
+
+    if prefill_plugin is None:
+        component["prefill"]["plugin"] = ""
+        altered = True
+    if prefill_attribute is None:
+        component["prefill"]["attribute"] = ""
+        altered = True
+
+    return altered
+
+
 CONVERTERS: dict[str, dict[str, ComponentConverter]] = {
```

Comment:
I was thinking we could add a `Literal` type alias somewhere referencing all the available types? e.g. `ComponentTypes: TypeAlias = Literal["textfield", ...]`.

It could be used as the key type here, but could also be useful for people to have reference

Reply:
I don't think the tradeoff of effort vs. benefit is worth it at this time.

The typescript type definitions will serve as The Reference when they're completed and should be the source of truth.

The python TypedDict approach has other drawbacks, preferably I'd go through a `dict -> dataclasses/pydantic -> dict -> JSON` flow so that we can benefit from python typing without suffering the typeddict limitations

Reply:
I see, but without having to deal with `TypedDict`, the use case here could be:

`CONVERTERS: dict[ComponentType, dict[str, ComponentConverter]] = {...}`

Reply:
And our own backend definitions, it would be great to have a similar type alias in the backend:

```py
AnyComponent = TextFieldComponent | CheckboxComponent | ...
```

and if possibly then infer `AnyComponent.type` from that, but I'm not even sure if that's technically possibly in python.

Perhaps we need to do it the other way around and define a literal for possibly types and constrains the base `Component` protocol/class using that. So all in all - too much uncertainty at the moment that needs to be cleared up first instead of haphazardly sprinkling types around over the entire codebase.

Reply:
> but I'm not even sure if that's technically possibly in python.

I'm afraid it isn't.

> too much uncertainty at the moment that needs to be cleared up first instead of haphazardly sprinkling types around over the entire codebase.

Yes, and dataclasses or dataclasses like libraries like pydantic could be used instead in the end (overwriting typedict fields is not allowed yet)